### DISCUSSION TITLE
Multi Tenant Support

### DISCUSSION THREAD

**Discussion by kaua-rob** [invertase/react-native-firebase#4964] [Ideas] [answered]
Hey guys,

I'm starting to develop an application with react-native-firebase-auth and I miss the multi-tenant support, to integrate with the multi-tenant Google Identity Platform.

This feature will be developed in future versions?

https://cloud.google.com/identity-platform/docs/multi-tenancy-quickstart

I saw the firebase android SDK docs, and I think it needs to map the "setTenantId" and "getTenantId" functions from native SDK.

https://firebase.google.com/docs/reference/android/com/google/firebase/auth/FirebaseAuth?authuser=0

**Comment by mikehardy** [comment]
I don't we have plans at current, though these things evolve.
Have you considered posting a PR to add it? I haven't looked at the relevant docs but the development experience for react-native-firebase has a very very efficient code/test loop with built-in support/use of the authEmulator even. After maybe 10-15mins of setup it might just be an hour or two of hackery then done?

https://github.com/invertase/react-native-firebase/blob/master/tests/README.md

**Comment by Madsim** [comment]
We need this feature as well. If I understand the auth flow correctly, it comes down to adding the tenantId to the "login call". Unfortunately we don't have any Java and/or Swing programmers at hand.

![tenant-request](https://user-images.githubusercontent.com/1423660/109401950-30ec3e00-7952-11eb-9068-e4f5abe105ea.jpg)

**Comment by mikehardy** [comment, marked as answer]
Hey @kmr299 @Madsim ! This is coming soon, there's a PR #5004 courtesy of @xianlinbox - it is nearly in shape to merge (I need to review again, but it is looking good) - you should subscribe to it and look for it in next release notes

**Reply by mikehardy** [reply]
@kmr299 @Madsim  This is out as of v11.1.0 - please test and if there's anything wrong let us know - we do not have multi-tenant set up in our test project and the auth emulator does not support multi-tenant so I was able to test it a fair bit, but not completely. That usually means there is something wrong, so I am listening intently on this one and would love to hear that people have tested this and after setting tenant id new users have their tenantId property set correctly
